airflow/providers/apache/druid/hooks/druid.py

```diff
@@ -149,6 +154,20 @@ def submit_indexing_job(
 
         self.log.info("Successful index")
 
+    def test_connection(self) -> tuple[bool, str]:
+        try:
+            conn = self.get_connection(self.druid_ingest_conn_id)
+            host = conn.host
+            port = conn.port
+            # ref : https://druid.apache.org/docs/latest/operations/api-reference/#tasks
+            response = requests.get(f"http://{host}:{port}/druid/indexer/v1/tasks")
```

Comment:
One of the main reasons for connection and test_connection method is storing the credentials securely and test them via this method, so IMHO this method should test if the credentials are valid or not. You can try something like:
```suggestion
            response = requests.get(
                f"http://{host}:{port}/druid/indexer/v1/tasks",
                 auth=self.get_auth(),
             )
```

Reply:
@hussein-awala 
The Druid Broker Hook overrides the DbApiHook, but the test connection is not implemented to use authentication information
https://github.com/apache/airflow/blob/main/airflow/providers/common/sql/hooks/sql.py#L550

So, I think adding authentication information to the Druid hook (broker, ingest) is handled in another issue